### PR TITLE
Recent versions of identity lib break the Rust FFI - EI-635

### PR DESCRIPTION
**Pull request by mihai-dinculescu**

### CODE REVIEW

**Comment by mirogta** [comment]
Great work, tests are passing.

What about the 152 warnings `warning: dereferencing a null pointer`?

E.g. [here](https://github.com/Iotic-Labs/iotics-identity-go/runs/4777128850?check_suite_focus=true#step:8:58)

**Comment by mihai-dinculescu** [comment]
> Great work, tests are passing.
> 
> What about the 152 warnings `warning: dereferencing a null pointer`?
> 
> E.g. [here](https://github.com/Iotic-Labs/iotics-identity-go/runs/4777128850?check_suite_focus=true#step:8:58)

Those are from the generated unit tests by bindgen. There's not much we can do about them.

**Review by mirogta — APPROVED** [review]
LGTM

**Review by smartrics — COMMENTED** [review]
just a minor comment 
Let me know what you think 
looks good to me otherwise (bar I can't really comment on rust code)

File: ffi/rust/src/lib.rs [hunk 1]
```diff
@@ -72,22 +209,34 @@ impl Drop for ffi_wrapper::CreateTwinDidWithControlDelegation_return {
...
-            go_string(&CString::new(config.agent_name.as_str())?),
-            go_string(&CString::new(config.agent_secret.as_str())?),
-            go_string(&CString::new(config.user_did.as_str())?),
-            go_string(&CString::new(config.audience.as_str())?),
+            agent_did.as_raw()?,
+            agent_key_name.as_raw()?,
+            agent_name.as_raw()?,
+            agent_secret.as_raw()?,
+            user_did.as_raw()?,
+            // using resolver_address as audience
```
[the remark this answers is not in the document]
Comment (mihai-dinculescu) [reply ffi/rust/src/lib.rs:225]: The snag is that this function doesn't have access to the host address as it's not required for the creation of the token (which can happily work across multiple hosts atm).

Would it be better to put a dummy value for now (e.g. "aud-not-in-use") until we figure out what we want to do with it?

[the remark this answers is not in the document]
Comment (smartrics) [reply ffi/rust/src/lib.rs:225]: No - ok; eventually we'll have to modify it since fixing aud will become a security concern

**Review by smartrics — APPROVED** [review]
approved - we'll need to fix aud at some point as per comment but until we do the check in the host, there's little point.

**Review by Evolter — APPROVED** [review]
LGTM

**Pull request merged**